src/spibus.c

```diff
@@ -0,0 +1,66 @@
+#include "autoconf.h"
+
+#include "board/gpio.h"   // gpio_out_write
+#include "board/irq.h"    // irq_disable
+#include "generic/spi.h"
+
+#include "basecmd.h"      // oid_alloc
+#include "command.h"      // DECL_COMMAND
+#include "sched.h"        // DECL_TASK
+
+struct spibus_t {
+    struct spi_config spi_config;
+    struct gpio_out pin;
+};
+
+#define BUFF_SIZE 32u
+static uint8_t spirespbuff[BUFF_SIZE] = {0};
+
+/* Config slave select pin and SPI bus */
+void command_config_spibus_ss_pin(uint32_t *args) {
+    struct spibus_t *spi =
+        oid_alloc(args[0],
+                  command_config_spibus_ss_pin,
+                  sizeof(*spi));
+    spi->pin        = gpio_out_setup(args[1], 1); // CS pin
+    spi->spi_config = spi_get_config(args[2], args[3]);
+}
+DECL_COMMAND(command_config_spibus_ss_pin,
+             "config_spibus_ss_pin oid=%c pin=%u spi_mode=%u spi_speed=%u");
+
+void command_spibus_transfer(uint32_t *args) {
+    struct spibus_t *spi =
+        oid_lookup(args[0], command_config_spibus_ss_pin);
+
+    uint8_t iter = 0;
+    uint8_t status = 1;
+    uint8_t len = args[1];
+
+    /* Read data from SPI slave */
+    uint8_t *msg = (uint8_t*)(uintptr_t)args[2];
+    if (len && len <= BUFF_SIZE) {
+        while (!spi_set_config(spi->spi_config));
+        gpio_out_write(spi->pin, 0); // Enable slave
+        do {
+            spirespbuff[iter++] =
+                spi_transfer((uint8_t)(*msg++));
```

Comment:
Any reason we are not using `spi_transfer_len` here?  Also is the buffer needed?  Can we not just place the result in data as the `command_send_spi_message` does?